jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/LogicalDeclarationComparator.java

```diff
@@ -0,0 +1,35 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.http.pathmap;
+
+import java.util.Comparator;
+
+/**
+ * Sort {@link MappedResource}s by their {@link MappedResource#getPathSpec()} logical declarations.
+ */
+public class LogicalDeclarationComparator implements Comparator<MappedResource>
```

Comment:
Can we think of a better name than LogicalDeclarationComparator? It's a comparator that works for MappedResources based on their PathSpecs .... is it a MappedResourceComparator? Is it a PathSpecComparator? In fact, could it be an anonymous class defined in PathMappings in the constructor for _mappings at line 49? I can't see where it's used anywhere else, but maybe it is?

Reply:
I think this class should go and be replaced with a lambda in the `PathMappings`' `TreeSet` constructor instead.

Reply:
But since this is @joakime 's code, maybe he had something else in mind?

Reply:
If you can make the tests pass then i'm good with any change you want to make.

Reply:
Okay, done.